samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,66 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    // Prepare an ancilla qubit in the |+> state.
+    use ancilla = Qubit();
+    H(ancilla);
+
+    let a = Mzz(control, ancilla);
+    let b = Mxx(ancilla, target);
+    let c = Mx(ancilla);
+    let c = Mz(ancilla);
+    if b == One {
+        Z(control);
+    }
+    if a != c {
+        X(target);
+    }
+}
+
+/// Prepare a Bell Pair.
+/// Source: [1] Figure 18a.
+operation BellPair(q1 : Qubit, q2 : Qubit) : Unit {
+    // Collapse the qubits to the X basis.
```

Comment:
nitpick: I think this might be collapsing to the Z basis, but there might be different conventions about how to describe this. Based on my understanding, I'd phrase this as:
```suggestion
    // Project each of the qubits onto the Pauli-Z basis.
```